src/main.c

```diff
@@ -270,6 +253,92 @@ static char *find_stdlib()
     return path;
 }
 
+static void add_imported_symbol(struct FileState *fs, const ExportSymbol *es)
+{
+    struct GlobalVariable *g;
+
+    switch(es->kind) {
+    case EXPSYM_FUNCTION:
+        for (AstToplevelNode *ast = fs->ast; ast->kind != AST_TOPLEVEL_END_OF_FILE; ast++) {
+            if ((ast->kind==AST_TOPLEVEL_DECLARE_FUNCTION || ast->kind==AST_TOPLEVEL_DEFINE_FUNCTION)
+                && !strcmp(ast->data.funcdef.signature.funcname, es->name))
+            {
+                // A function with this name will be declared/defined in the file.
```

Comment:
```suggestion
                // A function with this name will be declared/defined in the file.
                // Emit an error at the declaration/definition, because it comes after the import in the file.
                // We must do this here, because the declaration/definition was already processed.
```